rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/UnityAttachToEditorRunConfiguration.kt

```diff
@@ -92,6 +93,26 @@ class UnityAttachToEditorRunConfiguration(project: Project, factory: Configurati
         return null
     }
 
+    override fun checkRunnerSettings(
+        runner: ProgramRunner<*>,
+        runnerSettings: RunnerSettings?,
+        configurationPerRunnerSettings: ConfigurationPerRunnerSettings?
+    ) {
+        if (runner is DotNetDebugRunner) {
+            // If we're a class library project that isn't in a Unity project folder, we can't guess at the correct project
+            // to attach to, so throw an error and show the dialog. This value will be null until the backend has finished
+            // loading. However, because we're a Unity run configuration, we can safely assume we're a Unity project, and if
+            // we're not inside a Unity project folder, then we can't automatically attach, so throw an error and show the
+            // dialog
+
+            val isClassLibraryProject = project.isUnityClassLibraryProject()
```

Comment:
This needs a check for `pid != null` first. If we've previously attached to an editor, we try to reconnect. I think this logic is still true for class libraries.